### PR TITLE
Testing for OWL-6 and OWL-7

### PR DESCRIPTION
**Pull request by AppliNH**
+ By using a function which dials Auth gRPC to create a new user
+ Login and get token
+ Create a new chatroom and add our "main" testing user

### CODE REVIEW

File: server/chatroom/chatroom_server/server_test.go [hunk 1]
```diff
@@ -100,4 +196,64 @@ func TestGetChatroomsByUser(t *testing.T) {
...
+	}{
+		{
+			req: chatroompb.LeaveChatroomRequest{
+				Id: chatroomIdByToto,
+			},
+			want: chatroompb.LeaveChatroomResponse{
+				Success: true,
+			},
+		},
+	}
```
Comment (vgalin) [inline server/chatroom/chatroom_server/server_test.go:219]: We should also test that a user that is owner cannot leave its chatroom.

File: server/chatroom/chatroom_server/server_test.go [hunk 2]
```diff
@@ -100,4 +196,64 @@ func TestGetChatroomsByUser(t *testing.T) {
...
+	}{
+		{
+			req: chatroompb.DeleteChatroomRequest{
+				Id: chatroomIdByAppliNH,
+			},
+			want: chatroompb.DeleteChatroomResponse{
+				Success: true,
+			},
+		},
+	}
```
Comment (vgalin) [inline server/chatroom/chatroom_server/server_test.go:249]: We should also test that a user who is not a chatroom owner cannot delete the chatroom.

**Review by vgalin — CHANGES_REQUESTED** [review]
Adding two more tests would be nice.

**Comment by AppliNH** [comment]
I'm now using `reflect.DeepEqual` in the `assert` function. I believe it's better than simply using `!=`.
It also allowed me to test when the RPC was returning an error when it was supposed to.

More here : https://golang.org/src/reflect/deepequal.go

**Review by vgalin — APPROVED** [review]

**Pull request merged**